httpx/_content.py

```diff
@@ -150,11 +150,16 @@ def encode_urlencoded_data(
 
 
 def encode_multipart_data(
-    data: dict, files: RequestFiles, boundary: Optional[bytes] = None
+    data: dict,
+    files: RequestFiles,
+    boundary: Optional[bytes] = None,
+    content_type: Optional[str] = None,
 ) -> Tuple[Dict[str, str], MultipartStream]:
+    if content_type:
+        boundary = get_multipart_boundary_from_content_type(content_type)
```

Comment:
Shouldn't we check that `boundary is None` before computing `boundary`? (In other words, what happens when `content_type` and `boundary` do not agree on the value to use for `boundary`?)

Reply:
We're the only ones ever making the call into this function and we never call it with a mismatch. If I add the error I would either have to (1) add a test just to check the error or (2) pragma: no cover it. I'll add a comment instead.

Reply:
Shouldn't the `get_multipart_boundary_from_content_type` call be made upstream? e.g. in httpx/_models.py?

Reply:
I think _should_ would be a bit strong of a word, we certainly _could_ do that. I think the main reason for having it here is that `_models.py` doesn't know about multipart request or other encoding stuff, so it would be a bit of a violation of the layering that is currently set up to put something like `if content_type.startswith("multipart/form-data")`: <do multipart specific stuff>`.

Reply:
Yes, sorry, I went a bit strongly on it.

Reply:
No need to apologize for a good suggestion 😄 
Like I said I'm open to it, but I do feel that it breaks a bit with the way abstractions are currently set up in the codebase